cpp/Platform.Delegates/Platform.Delegates.Delegate.h

```diff
@@ -16,54 +14,134 @@ namespace Platform::Delegates
     template <typename ReturnType, typename... Args>
     class Delegate<ReturnType(Args...)>
     {
+    public:
         using DelegateRawFunctionType = ReturnType(Args...);
         using DelegateFunctionType = std::function<DelegateRawFunctionType>;
 
+        constexpr Delegate() noexcept = default;
+
+        Delegate(const Delegate&) noexcept = default;
+
+        Delegate(Delegate&&) noexcept = default;
+
+        constexpr Delegate(DelegateRawFunctionType simpleFunction) noexcept
+                : simpleFunction(simpleFunction) {}
+
+        Delegate(const DelegateFunctionType &complexFunction)
+                : complexFunction(std::make_shared<DelegateFunctionType>(complexFunction)) {}
```

Comment:
Is it safe to make `shared_ptr` from reference to `std::function`?

Reply:
Yes, because here we are making shared_ptr with a copy of the std::function, not a copy of the reference

Reply:
If we use copy anyway, my be we can store it directly in the class' field? Why we need a `shared_ptr` on a copy of the `std::function`?

Reply:
this is better because we also store pointers to a MemberMethod or RawFunction object. In an amicable way, in the next pull request, you need to replace three pointers with one std::variant to save memory, and unify the interaction with data inside the delegate. Of course, there is a problem with the fact that additional indirections and atomic variables in shared_pointer can slow down the code, but this is not the bottleneck, given that mutexes are used outside for synchronization

Reply:
Wow, good idea.